### ISSUE TITLE
ConstExpression

### ISSUE THREAD

**Issue by kant2002** [ForNeVeR/Cesium#56]
After I add some simple lowering support, I plan to add prefix ++ operator which in standard defined as `E++ <==> E = E+1`. But that show limitation of the current approach to parsing. AST for ConstExpression directly tied to token to get value, but I need just value without token attached (or that token is `++`). 

I do not want perform any surgery/changes here unless we both on same page about goals. I play with optional, `int Value` property to `ConstExpression` but that approach seems to be not ideal. I think whole `ConstExpression` should be reworked to something like this
```
enum ConstantKind
{
    Int,
    Float,
    Double,
    Char,
    CharArray, // or String
}
record ConstExpressions(ConstantKind kind, object Value, ICToken? token);
```
Other approach would be to have separate `IntConstantExpression`, `DoubleConstantExpression`, `FloatConstantExpression`, etc. I think this less desirable approach since that's require constant micromanagement for types in expressions.

I honestly do not see proper "small" change at this point.

**Comment by ForNeVeR** [comment]
From these two ideas, I would lean towards separate types for each expression (type safety first!), i.e. the one about `IntConstantExpression`, `DoubleConstantExpression`, `FloatConstantExpression`, etc.

Though the former one sounds reasonable, too.

I have plans to create a separate and better structured IR (if you wish to call it that) on top (or even instead of?) the original C AST, with more useful extensions like `constExpression.GenerateCode()` maybe.

For now, if you wanna get yourself busy, feel free to use either approach, both are okay-ish.

**Comment by ForNeVeR** [comment]
I guess this was implemented in #60.
